airflow/providers/amazon/aws/operators/glacier.py

```diff
@@ -54,3 +54,55 @@ def __init__(
     def execute(self, context: Context):
         hook = GlacierHook(aws_conn_id=self.aws_conn_id)
         return hook.retrieve_inventory(vault_name=self.vault_name)
+
+
+class GlacierUploadArchiveOperator(BaseOperator):
+    """
+    This operation adds an archive to a vault
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:GlacierUploadArchiveOperator`
+
+    :param account_id: The AWS account ID of the account that owns the vault.
+        You can either specify an AWS account ID or optionally a single `-` (hyphen),
+        in which case Amazon S3 Glacier uses the AWS account ID associated with
+        the credentials used to sign the request
+    :param vault_name: The name of the vault
+    :param body: A bytes or seekable file-like object. The data to upload.
```

Comment:
Noting that my concern of https://github.com/apache/airflow/pull/22758#discussion_r843298615 is also applied here. The question of where the bytes come from is very important. If it's from Xcom then body probably need to be templated (but I guess we don't really want users to push it to Xcom?) if it's from somewhere else then we probably a transfer operator is more suitable.
To me it seems like the only use case for this operator is when the body is hard coded and I'm not sure how useful that is?

That said... If people find this useful as is I'm OK with it.

Reply:
If the source (where data is coming from) is known then it makes sense to have a transfer operator but my purpose here is to automate the frequently used python function in DAG so that it can be reused as an operator. It would be useful if I want to upload my local data to Glacier or data from any source storage for which we do not have native API (download data and upload)

Reply:
That is still transfer operator: LocalTo* we have several of those in the project.
The issue is that operators don't share disk so the only valid case for upload from disk without transfer operator is if the task you are running is doing source to destination in single task (usually PythonOperator).

As I said I'm OK with adding this but I didn't see any real production case usage for this kind of operators. If this is one and can be shown in an example DAG I be happy to learn something new :)